### PR TITLE
Typescript+Axios: Separate model and api classfiles and package

### PR DESCRIPTION
**Pull request by mvniekerk**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `4.0.x`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @nicokoenig (2018/09) @topce (2018/10)

### Description of the PR
This PR needs to follow/branches https://github.com/OpenAPITools/openapi-generator/pull/2002

What it does:
 - Optionally splits API and Model class files for Axios in  the apiPackage and modelPackage directory
 - Filenames are kebab case (MyAwesomeClass -> my-awesome-class.ts)

### CODE REVIEW

**Comment by macjohnny** [comment]
please request review once #2002 is merged

**Comment by mvniekerk** [comment]
> please request review once #2002 is merged

Will do

**Comment by wing328** [comment]
https://github.com/OpenAPITools/openapi-generator/pull/2002 has been merged into master.

@mvniekerk can you resolve the merge conflicts when you've time? let me know if you need any help on that.

**Comment by mvniekerk** [comment]
@wing328 @macjohnny - made it mergable again.

File: bin/typescript-axios-petstore-with-npm-version-and-separate-models-and-api.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#!/bin/sh
+
+SCRIPT="$0"
+echo "# START SCRIPT: $SCRIPT"
+
+while [ -h "$SCRIPT" ] ; do
```
Comment (macjohnny) [inline bin/typescript-axios-petstore-with-npm-version-and-separate-models-and-api.sh]: please add a windows-version, too

Comment (mvniekerk) [reply bin/typescript-axios-petstore-with-npm-version-and-separate-models-and-api.sh]: Addressed in 098971da88

**Comment by mvniekerk** [comment]
Not to be pushy -
Kindly accept or reject this PR, please. I'll be leaving Grindrod Bank end of March and my access to their forked repository is not a given post March 2019. Would like to wrap up and clean up.
CC @macjohnny @wing328

**Comment by macjohnny** [comment]
@mvniekerk thanks a lot for your effort!

**Comment by macjohnny** [comment]
@nicokoenig could you please have a quick look at this one?

**Comment by mvniekerk** [comment]
@mac

> IMHO it looks good. There are only minimal, non-breaking changes in the existing samples. I haven't tested the new samples with separated model and api, but at first glance this looks good, too.
> @mvniekerk can you confirm that you tested the newly generated files in an application?

Hey @macjohnny
Yes - running in production code

**Comment by macjohnny** [comment]
@wing328 I think this should be good to go

**Pull request merged**